DocumentationToolingNotes.md

```diff
@@ -0,0 +1,62 @@
+
+There are four relevant d.ts files powering different subsections of the docs.
+
+- [office-js/index.d.ts](https://raw.githubusercontent.com/DefinitelyTyped/DefinitelyTyped/master/types/office-js/index.d.ts) (The Release definitions.)
+  - [Excel (Release)](https://docs.microsoft.com/javascript/api/excel_release)
+  - [OneNote](https://docs.microsoft.com/javascript/api/onenote)
+  - [PowerPoint](https://docs.microsoft.com/javascript/api/visio)
+  - [Visio](https://docs.microsoft.com/javascript/api/visio)
+  - [Word (Release)](https://docs.microsoft.com/javascript/api/word_release)
+  - [OfficeExtensions subsection of the Common API](https://docs.microsoft.com/javascript/api/office)
+- [office-js-preview/index.d.ts](https://raw.githubusercontent.com/DefinitelyTyped/DefinitelyTyped/master/types/office-js-preview/index.d.ts) (The Preview definitions.)
+  - [Excel (Preview)](https://docs.microsoft.com/javascript/api/excel)
+  - [Outlook (Preview)](https://docs.microsoft.com/javascript/api/outlook)
+  - [Word (Preview)](https://docs.microsoft.com/javascript/api/word)
+  - [Common API](https://docs.microsoft.com/javascript/api/office)
+- [custom-functions-runtime](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/custom-functions-runtime/index.d.ts)(The Excel Custom Functions runtime definitions.)
+  - [Custom Functions](https://docs.microsoft.com/javascript/api/custom-functions-runtime)
+- [office-runtime](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/office-runtime/index.d.ts)(The office runtime definitions for the Custom Functions platform.)
+  - [Office Runtime](https://docs.microsoft.com/javascript/api/office-runtime)
+
+Legacy versions of the APIs have their own d.ts files. These are preserved when a new API set is released. They can also be generated using the [Version Remover tool](https://github.com/OfficeDev/office-js-docs-reference/blob/master/generate-docs/tools/VersionRemover.ts). These old d.ts files are maintained so that in the event APIs are patched or altered, the original behavior is still documented. This is useful if you have to target an older version of the API.
+
+### Code snippets
+
+Code example snippets are added to the reference pages from two sources:
+
+- [Script Lab Samples](https://github.com/OfficeDev/office-js-snippets)
+- [Local Code Snippets](https://github.com/OfficeDev/office-js-docs-reference/tree/master/docs/code-snippets)
+
+The local snippets are in host-specific yaml files. Their content is organized by class and field, so it can be mapped to the appropriate place in a reference page. The language of the snippet (JavaScript or TypeScript) is inferred by the use of await statements.
```

Comment:
Just as FYI JavaScript also uses await, and we should be using that pattern as it is considered "modern JavaScript". This could be a bug in how to recognize the language type.

Reply:
I'll file an issue to address this.